### PR TITLE
card title: prevent space and no text

### PR DESCRIPTION
**Pull request by jakobroehrl**
Signed-off-by: Jakob Röhrl <jakob.roehrl@web.de>

* Resolves: #589 
* Target version: master

### CODE REVIEW

**Comment by jakobroehrl** [comment]
@juliushaertl 
Do you have an idea why this is working here https://github.com/nextcloud/deck/blob/7c55d052e25ff49870426b0548e70eac9d9d7f6b/src/components/board/Stack.vue#L86

but here not: https://github.com/nextcloud/deck/blob/7c55d052e25ff49870426b0548e70eac9d9d7f6b/src/components/cards/CardItem.vue#L49

**Comment by juliushaertl** [comment]
The form will only be checked if submitted not when a regular button is used with a click event 
https://github.com/nextcloud/deck/blob/7c55d052e25ff49870426b0548e70eac9d9d7f6b/src/components/cards/CardItem.vue#L51

File: src/components/card/CardSidebar.vue [hunk 1]
```diff
@@ -464,6 +464,9 @@ export default {
 			this.descriptionSaving = false
 		},
 		updateTitle(newTitle) {
+			if (newTitle === '' || newTitle === ' ') {
```
Comment (juliushaertl) [inline src/components/card/CardSidebar.vue:467]:
```suggestion
			if (newTitle.trim() === '') {
```

File: src/components/card/CardSidebar.vue [hunk 2]
```diff
@@ -464,6 +464,9 @@ export default {
...
+				return
```
Comment (juliushaertl) [inline src/components/card/CardSidebar.vue:468]: Would be nice to have some error handling, though I'm not sure the sidebar currently supports this, so maybe just show a toast with an error message.

**Comment by stefan-niedermann** [comment]
So you are only limiting it in the UI - please keep in mind that performing this validation at the backend, it will be a breaking change :)

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**